File: logic-exercises/low-level/exe1.c
#include <stdio.h>
#include <stdbool.h>

int main(void) {
    while(true) {
        int answer;
        printf("Insira a senha: ");
        scanf("%d", &answer);

        if (answer == 2002) {
            printf("Acesso permitido!");
            return 0;
        } else {
            printf("Senha invalida!\n");
        }
    }
    return 0;
}

File: logic-exercises/eclipse/exe1.java
package eclipse;

import java.util.Scanner;

public class exe1 {
    public static void main(String[] args) {
        while (true) {
            Scanner scan = new Scanner(System.in);
            System.out.println("Insira a senha: ");
            String answer = scan.nextLine();

            if (answer.equals("2002")) {
                System.out.println("Acesso permitido!");
            } else {
                System.out.println("Senha inválida!");
            }
            scan.close();
            return;
        }
    }
}